module/PowerShellEditorServices/Start-EditorServices.ps1

```diff
@@ -246,6 +268,37 @@ function Set-NamedPipeMode {
 LogSection "Console Encoding"
 Log $OutputEncoding
 
+function Test-NamedPipeName-OrCreate-IfNull {
+    param(
+        [string]
+        $PipeName
+    )
+    if (-not $PipeName) {
+        $PipeName = New-NamedPipeName
+    }
+    else {
+        if (-not (Test-NamedPipeName -PipeName $PipeName)) {
```

Comment:
So what happens when Test-NamedPipeName equals `true`? Will it return `null`? Should that ever happen?

I feel as if the Test & Create part should be two separate functions.

Reply:
The function will fail and abort with the error if the supplied by the user pipe name is not a valid name. If a valid name was supplied it just returns this pipe name. If no pipe name was provided it creates it.